### ISSUE TITLE
[BUG] NaN Loss in Dipole Fitting with DeepMD-kit 3.0.1

### ISSUE THREAD

**Issue by frssp** [deepmodeling/deepmd-kit#4620]
## Bug Description
When using atomic dipole fitting in DeepMD-kit 3.0.1, the training loss becomes NaN. The same example works correctly in version 2.2.10.

## Steps to Reproduce
1. Use the example in the official repository: `examples/water/dplr/train`
2. Run `dp train dw.json`

## Error Output
DeepMD-kit 3.0.1:
Std out
```
...
[2025-02-27 13:31:33,793] DEEPMD INFO    batch     100: trn: rmse = 0.00e+00, rmse_lc =      nan, lr = 1.00e-18
[2025-02-27 13:31:33,793] DEEPMD INFO    batch     100: total wall time = 0.94 s
[2025-02-27 13:31:34,151] DEEPMD INFO    batch     200: trn: rmse = 0.00e+00, rmse_lc =      nan, lr = 1.00e-18
[2025-02-27 13:31:34,151] DEEPMD INFO    batch     200: total wall time = 0.36 s
...
```

DeepMD-kit 2.2.10 (working correctly):

cat lcurve.out
```
#  step      rmse_trn   rmse_lc_trn         lr
# If there is no available reference data, rmse_*_{val,trn} will print nan
      0      2.30e-01      2.30e-01    1.0e-18
    100      2.50e-01      2.50e-01    1.0e-18
    200      2.48e-01      2.48e-01    1.0e-18
    300      2.25e-01      2.25e-01    1.0e-18
    400      2.26e-01      2.26e-01    1.0e-18
...
```

### DeePMD-kit Version

DeePMD-kit v3.0.1

### Backend and its version

cpu, tf.__version__ '2.14.1', pytorch 

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

examples/water/dplr/train

It also fails other inputs using dipole.

### Steps to Reproduce

cd {dp_src}/examples/water/dplr/train
dp train dw.json

### Further Information, Files, and Links

Currently Loaded Modulefiles:
 1) cuda_11.3.0   2) cudnn-11.x-linux-x64-v8.1.1.33   3) nccl_2.9.8-1+cuda11.3   4) gcc-10.2.0   5) openmpi-4.1.0-cu113-ucx1110   6) mkl  

but. also fails with the cpu version of deepmd-kit v.3.0.1 installed using pip.

**Comment by njzjz** [comment]
See #4536.
